libcontainer/nsenter/nsexec.c

```diff
@@ -713,10 +715,12 @@ void send_idmapsources(int sockfd, pid_t pid, char *idmap_src, int idmap_src_len
 		ret = sys_mount_setattr(fd_tree, "", AT_EMPTY_PATH, &attr, sizeof(attr));
 		if (ret < 0) {
 			sane_kill(pid, SIGKILL);
-			if (errno == EINVAL)
-				bail("failed to change mount attributes, maybe the filesystem doesn't supports ID-mapped mounts");
+			if (errno == ENOSYS)
+				bail("failed to use mount_setattr(2), the kernel doesn't support ID-mapped mounts");
+			else if (errno == EINVAL)
+				bail("failed to use mount_setattr(2), maybe the filesystem doesn't support this ID-mapped mount: %s, error msg", idmap_src);
```

Comment:
Perhaps use the same form as above, i.e. "with path: %s"?

Plus the same comment as above.

Reply:
Good call, and I can remove the ", error msg" too then. I think it is clear without it now